src/devcenter/azext_devcenter/utils.py

```diff
@@ -88,4 +95,15 @@ def get_delayed_time(delay_time, action_time):
     split_time = delay_time.split(":")
     hours = int(split_time[0])
     minutes = int(split_time[1])
-    return action_time + timedelta(hours=hours, minutes=minutes)
+    delayed_time = action_time + timedelta(hours=hours, minutes=minutes)
+    return delayed_time.strftime("%Y-%m-%dT%H:%M:%S.%fZ")
+
+
+def get_dataplane_endpoint(cli_ctx, endpoint=None, dev_center=None, project_name=None):
+    validate_endpoint(endpoint, dev_center)
+    if endpoint is None and dev_center is not None:
+        project = get_project_data(cli_ctx, dev_center, project_name)
+        endpoint = project["devCenterUri"]
+    endpoint = endpoint.split('//', 1)[-1]
```

Comment:
It seems the user can defined data-plane endpoints by three ways:

- directly provide the endpoint
- provide the `dev_center` and using the first devCenterUri of the project
- provide both `dev_center` and `project_name` and using the devCenterUri

My question is why need the third way if providing the `dev_center` only can get the devCenterUri in response.

And for the second way actually you can config the data plane client like this which will retrieve the `properties.devCenterUri` as endpoint from devcenter resource from the control plane api.
![image](https://github.com/user-attachments/assets/49df561a-351a-44fb-bdc1-f2f98429be9d)